lib/knock/workflows.rb

```diff
@@ -63,6 +63,91 @@ def cancel(key:, cancellation_key:, recipients: nil)
 
         execute_request(request: request)
       end
+
+      # Creates schedule instances for all recipients for the given
+      # workflow.
+      #
+      # @param [String] key The workflow key
+      # @param [String, Hash] actor An optional actor identifier to be used when trigger the workflow
+      # @param [Array<String, Hash>] recipients The recipient identifiers
+      # @param [Array<Hash>] repeats The repeat rules for when to schedule the workflow run
+      # @param [Hash] data Parameters to be used as variables on the workflow run
+      # @param [String] tenant An optional tenant identifier
+
+      def create_schedules(key:, actor:, recipients:, repeats:, scheduled_at: nil, data: {}, tenant: nil)
+        attrs = {
+          workflow: key,
+          actor: actor,
+          recipients: recipients,
+          repeats: repeats,
+          data: data,
+          tenant: tenant,
+          scheduled_at: scheduled_at
+        }
+
+        request = post_request(
+          auth: true,
+          path: '/v1/schedules',
+          body: attrs
+        )
+
+        execute_request(request: request)
+      end
+
+      # Updates schedule instances with the attributes passed as params
+      #
+      # @param [Array<String>] schedule_ids The ids of the schedules to be updated
+      # @param [Hash] schedule_attrs The attributes to be used to update the schedules
+      #   Possible attributes: actor, repeats, data and tenant
+
+      def update_schedules(schedule_ids:, schedule_attrs:)
```

Comment:
Hmm should this more closely mirror the `create_schedules` and we don't put these attrs under `schedule_attrs` but put them on the top level?

Reply:
@cjbell I tried doing that, but some of the attributes are optional for updating schedules, and we can't use defaults because that may update the current attributes for a schedule. 

```
 def update_schedules(schedule_ids:, actor: nil ...rest of attributes)
 ```
 If the user does not intend to update the actor, we would do it.
 If we decide to check if the actor value is not nil, we won't allow updating the actor to be null which should be possible.

Reply:
Ahhh I hear ya – ok!